### PR TITLE
add ability to specify branches - fixes #9

### LINKED ISSUE #9: ability to specify branches
It'd be great if you could specify a branch to use for searching in the config. It'd be even better if you could do it on the fly in the UI

### PR DESCRIPTION
**Pull request by caglar10ur**

### CODE REVIEW

**Comment by jklein** [comment]
This LGTM, @kellegous take a look when you get a chance and feel free to merge.

File: config-example.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
             "url" : "https://www.github.com/YourOrganization/RepoOne.git"
         },
         "AnotherRepo" : {
-            "url" : "https://www.github.com/YourOrganization/RepoTwo.git"
+            "url" : "https://www.github.com/YourOrganization/RepoTwo.git",
+            "branch: "v1"
```
Comment (alex) [inline config-example.json]: Missing closing quote here.

Comment (caglar10ur) [inline config-example.json]: Corrected, thanks!

**Comment by kellegous** [comment]
@caglar10ur As I was trying this out, I found myself not wanting to have to define a branch as a new repo. I would rather have the following config:

```
        "Hound" : {
            "url" : "https://www.github.com/etsy/Hound.git"
           "branches" : ["master", "feature-a"]
        }
```

This also makes it easier to avoid cloning the same repo twice, which we would definitely not want to do with some of our Etsy repos.

**Comment by caglar10ur** [comment]
@kellegous hmm, make sense. I'll update the PR with the suggested change.

**Comment by caglar10ur** [comment]
@kellegous refreshed the PR. It seems to be working here...

**Comment by belak** [comment]
Doesn't this still make multiple clones per repo? Is there any way to leave the branch out of cloning/pulling and just use Checkout when we need to update the indexes for that branch?

**Comment by caglar10ur** [comment]
Nope, unless I miss something, it does exactly what you say. It clones once and then checkouts to branches whenever needed.

**Comment by belak** [comment]
Oh, I see. My mistake.

**Comment by jklein** [comment]
@caglar10ur can you fix the merge conflict and we'll take another look?

**Comment by caglar10ur** [comment]
@jklein sure, will try...

**Comment by caglar10ur** [comment]
@jklein pinging just in case you missed this

**Comment by jklein** [comment]
@caglar10ur this seems pretty reasonable to me. Let's see what @kellegous thinks.

**Comment by kellegous** [comment]
I'm going to take a look at this today. Sorry to take so long, I've been on the run up to a product launch and falling behind in these reviews.

**Comment by kellegous** [comment]
There are a couple of issues that I'd like to fix before merging. Maybe I can help with these.
 1. The branches are not shown in the repo name in the UI.
 2. The repo selection dropdown is busted (I think due to the name collision on the repo names).

Maybe I'll merge these changes into a branch off of master to fix these. Does that work?

**Comment by caglar10ur** [comment]
Yes, please feel free to do the neccesary changes as I'm not sure whether I'll be able to address those UI issues soon

**Comment by rhbecker** [comment]
My org just started playing with hound (it's awesome!) and pretty immediately starting wishing for more flexibility around indexing of multiple branches.

Is this effort dead? If so, is it because ...

1. original authors lost interest?
2. hound owners didn't value the use case?
3. work arounds were found?
4. something else?

**Comment by jklein** [comment]
@rhbecker I think @kellegous and I have just been super busy and haven't taken the time to get this over the goal line. If we can get a clean PR that will merge and addresses the points in Kelly's last comment I think we're good to go.

**Comment by mikepurvis** [comment]
I'm very interested in this fix— our standard branch names are `devel`, so having this would be a requirement for us (being able to specify tag names for indexing past releases will be a side effect of this which would also be great).

**Comment by kellegous** [comment]
This will require a restart, I'm going to close for now.

**Comment by dmsimard** [comment]
https://github.com/etsy/hound/pull/275 provides the ability to specify a git branch other than the hardcoded "master" and it works for us.

**Pull request closed without merging**